version: '3.8'

services:

  flask-app:
    image: flask-app
    container_name: flask-app
    build:
      context: .
      dockerfile: ./app.Dockerfile
    ports:
      - 5000:5000
    environment:
      FLASK_DEBUG: 'True'
      DATABASE_URI: postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${POSTGRES_HOST}:5432/${POSTGRES_DB}
    networks:
      - flask-app-network
    restart: on-failure
    depends_on:
      - database
    deploy:
      replicas: 1

  database:
    image: postgres:14
    env_file: .env
    environment:
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_DB=${POSTGRES_DB}
    networks:
      - flask-app-network
    restart: always

networks:
    flask-app-network:
      driver: bridge
      name: flask-app-network
